Assets/Scripts/CombatZone.cs

```diff
@@ -33,50 +38,97 @@ public void Start()
     {
         if (CombatStarted)
             return;
 
         if (other.gameObject.CompareTag("Player"))
         {
+            Timer.StartTimer(MovementTime);
+            StartCoroutine(StartingCombat());
             InitiateCombat();
         }
     }
 
     private void InitiateCombat()
     {
         CombatStarted = true;
+        InitalPositionsEnemies = new Vector3[Enemies.Length];
+        InitalPositionsPlayers = new Vector3[Players.Length];
 
-        SetCombatPositions();
+        for (int i = 0; i < InitalPositionsEnemies.Length; i++)
+        {
+            InitalPositionsEnemies[i] = Enemies[i].transform.position;
+            Enemies[i].GetComponent<Combatant>().EnterCombat();
+        }
 
-        CombatSystem.StartCombat(this.gameObject, Players, Enemies);
+        for (int i = 0; i < InitalPositionsPlayers.Length; i++)
+        {
+            Players[i].GetComponent<Combatant>().EnterCombat();
+            InitalPositionsPlayers[i] = Players[i].transform.position;
+        }
     }
 
-    private void SetCombatPositions()
+    private void DisablePlayerMovement()
+    {
+        if (Timer.IsFinished())
+            return;
+
+
+        for (int i = 0; i < PlayerPositions.Length; i++)
+        {
+            GameObject player = Players[i];
+            player.GetComponent<Player>().DisableMovement();
+            player.SetActive(false);
+            player.SetActive(true);
+        }
+    }
+
+    public void SetCombatPositionsUpdate()
     {
         for (int i = 0; i < EnemyPositions.Length; i++)
         {
-            GameObject enemy = Enemies[i];
-            enemy.transform.position = EnemyPositions[i].transform.position;
-            enemy.GetComponent<Combatant>().EnterCombat();
+            Enemies[i].transform.position = Vector3.Lerp(InitalPositionsEnemies[i],
+                                                         EnemyPositions[i].transform.position,
+                                                         Timer.GetProgress() / 2f);
+
+            Vector3 direction = (EnemyPositions[i].transform.position - Enemies[i].transform.position).normalized;
+            Enemies[i].transform.rotation = Quaternion.Lerp(Enemies[i].transform.rotation,
+                                                            Quaternion.LookRotation(direction), 
+                                                            Timer.GetProgress() / 2f);
         }
 
-        for (int j = 0; j < PlayerPositions.Length; j++)
+        for (int i = 0; i < PlayerPositions.Length; i++)
         {
-            GameObject player = Players[j];
-            player.GetComponent<Combatant>().EnterCombat();
-            player.GetComponent<Player>().DisableMovement();
+            Players[i].transform.position = Vector3.Lerp(InitalPositionsPlayers[i],
+                                                         PlayerPositions[i].transform.position,
+                                                         Timer.GetProgress() / 2f);
```

Comment:
that 2f is hardcoded

Reply:
Whoops good catch. Forgot to change that.